### PR TITLE
refactor yarn spawn

### PR DESCRIPTION
**Pull request by sbonasu**
Summary

Instead of glob use `yarn workspaces list` to return workspaces. This is more accurate than globs

Test Plan
All tests are passing but some are timing out with default timeout

Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

### CODE REVIEW

File: src/build-dep-graph.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const path = require('path');
 const semver = require('semver');
 const dependencyTypes = require('./dependency-types');
 const readJson = require('./json').read;
-const getWorkspacesPaths = require('./get-workspaces-paths');
+const getWorkspaces = require('./get-workspaces-paths').getWorkspacesPaths;
```
Comment (kellyselden) [inline src/build-dep-graph.js:7]: `const { getWorkspacesPaths } = require('./get-workspaces-paths');` ?

File: test/release-test.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ describe(_release, function() {
     });
   }
 
-  it('works', async function() {
+  it.only('works', async function() {
```
Comment (kellyselden) [inline test/release-test.js]: This seems like an accident.

File: src/get-workspaces-paths.js [hunk 3]
```diff
@@ -20,21 +18,22 @@ async function getWorkspacesPaths({
...
   }
-    
-  workspaces = [...new Set(_1dFilesArray)];
-    
+
   return workspaces;
 }
 
-module.exports = getWorkspacesPaths;
+module.exports = {
```
Comment (kellyselden) [inline src/get-workspaces-paths.js:37]: Maybe this should be in the other PR?

File: src/get-workspaces-paths.js [hunk 4]
```diff
@@ -1,8 +1,6 @@
 'use strict';
 
 const execa = require('execa');
-const { promisify } = require('util');
-const glob = promisify(require('glob'));
```
Comment (kellyselden) [inline src/get-workspaces-paths.js:5]: Are we using this elsewhere? Should we remove it from package.json?

File: src/get-workspaces-paths.js [hunk 5]
```diff
@@ -20,21 +18,22 @@ async function getWorkspacesPaths({
       .split(/\r?\n/)
       .map(workspace => path.relative(workspaceCwd, workspace));
 
+    workspaces = [...new Set(_1dFilesArray)];
```
Comment (kellyselden) [inline src/get-workspaces-paths.js:21]: I wonder if this is needed anymore?

File: src/get-workspaces-paths.js (hunk 4 above)
Comment (sbonasu) [reply src/get-workspaces-paths.js:5]: nope, I removed it

File: src/get-workspaces-paths.js (hunk 5 above)
Comment (sbonasu) [reply src/get-workspaces-paths.js]: not needed, removed

File: src/build-dep-graph.js (hunk 1 above)
Comment (sbonasu) [reply src/build-dep-graph.js]: I dropped commit to refactor code in src/build-dep-graph.js, I will do that in other PR

File: test/release-test.js (hunk 2 above)
Comment (sbonasu) [reply test/release-test.js]: ah, it sneaked in this commit but overall doesn't show in diff

Comment (sbonasu) [reply test/release-test.js]: resolved

**Pull request merged**